index.js

```diff
@@ -0,0 +1,954 @@
+                 keys.every(function(k) { return equals(other[k], self[k]); });
+        },
+
+        //  Object#concat :: StrMap a ~> StrMap a
+        'fantasy-land/concat': function(strMap) {
+          var result = {};
+          for (var k in this) result[k] = this[k];
+          for (k in strMap) result[k] = strMap[k];
+          return result;
+        },
+
+        //  Object#map :: StrMap a ~> (a -> b) -> StrMap b
+        'fantasy-land/map': function(f) {
+          var result = {};
+          for (var k in this) result[k] = f(this[k]);
+          return result;
+        },
+
+        //  Object#reduce :: StrMap a ~> ((b, a) -> b, b) -> b
+        'fantasy-land/reduce': function(f, initial) {
+          var result = initial;
+          for (var k in this) result = f(result, this[k]);
+          return result;
+        }
+      }
+    },
+
+    Function: {
+      prototype: {
+        //  Function#toString :: Function ~> () -> String
+        toString: function() { return String(this); },
+
+        //  Function#equals :: Function ~> Function -> Boolean
+        'fantasy-land/equals': function(other) { return other === this; },
+
+        //  Function#map :: (a -> b) ~> (b -> c) -> (a -> c)
+        'fantasy-land/map': function(f) {
+          var functor = this;
+          return function(x) { return f(functor(x)); };
+        },
+
+        //  Function#promap :: (b -> c) ~> (a -> b, c -> d) -> (a -> d)
+        'fantasy-land/promap': function(before, after) {
+          var profunctor = this;
+          return function(x) { return after(profunctor(before(x))); };
+        },
+
+        //  Function#ap :: (a -> b) ~> (a -> b -> c) -> (a -> c)
+        'fantasy-land/ap': function(a2b2c) {
+          var a2b = this;
+          return function(x) { return a2b2c(x)(a2b(x)); };
+        }
+      }
+    }
+
+  };
+
+  //# toString :: Showable a => a -> String
+  //.
+  //. Returns a useful string representation of its argument. The Showable
+  //. constraint indicates that applying `String` to the argument must not
+  //. raise an exception. An example of a value for which this is not true
+  //. is `Object.create(null)`.
+  //.
+  //. Dispatches to the argument's `toString` method if appropriate.
+  //.
+  //. Where practical, `equals(eval(toString(x)), x) = true`.
+  //.
+  //. ```javascript
+  //. > toString(-0)
+  //. '-0'
+  //.
+  //. > toString(['foo', 'bar', 'baz'])
+  //. '["foo", "bar", "baz"]'
+  //.
+  //. > toString({x: 1, y: 2, z: 3})
+  //. '{"x": 1, "y": 2, "z": 3}'
+  //. ```
+  var toString = (function() {
+    //  $seen :: Array Any
+    var $seen = [];
+
+    return function(x) {
+      if ($seen.indexOf(x) >= 0) return '<Circular>';
+
+      var xType = type(x);
+      var toStringMethod =
+        has(xType, implementations) &&
+        has('toString', implementations[xType].prototype) ?
+          implementations[xType].prototype.toString :
+          x.toString;
```

Comment:
If user overrides toString on some type for which there is implementation `implementations ` then it would not be called.

I think it it's not intentional, we could store default implementations in `defaultImplementations` for example and here check if `x.toString` is same as one from `defaultImplementations` (if exists) then call `implementations` as default one is not overridden but if it's overridden then call `x.toString`. 

(we don't have this issue in `T.getBoundMethod` as it's firs looking method on instance)